x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/user/TransportSetEnabledAction.java

```diff
@@ -62,4 +64,16 @@ protected void doExecute(Task task, SetEnabledRequest request, ActionListener<Ac
             listener.delegateFailure((l, v) -> l.onResponse(ActionResponse.Empty.INSTANCE))
         );
     }
+
+    private boolean isSameUserRequest(SetEnabledRequest request) {
+        final var effectiveSubject = securityContext.getAuthentication().getEffectiveSubject();
+        final var realmType = effectiveSubject.getRealm().getType();
+        if (ReservedRealm.TYPE.equals(realmType) == false && NativeRealmSettings.TYPE.equals(realmType) == false) {
```

Comment:
Usernames are mutually exclusive across these two realms so we don't have to worry about native user trying to disable same-named reserved user.